webknossos/webknossos/dataset/_utils/buffered_slice_writer.py

```diff
@@ -238,29 +231,77 @@ def _flush_buffer(self) -> None:
 
             raise exc
         finally:
-            self.slices_to_write = []
+            self._slices_to_write = []
 
     def _get_slice_generator(self) -> Generator[None, np.ndarray, None]:
         current_slice = 0
         while True:
             data = yield  # Data gets send from the user
-            if len(self.slices_to_write) == 0:
-                self.buffer_start_slice = current_slice
+            if len(self._slices_to_write) == 0:
+                self._buffer_start_slice = current_slice
             if len(data.shape) == 2:
                 # The input data might contain channel data or not.
                 # Bringing it into the same shape simplifies the code
                 data = np.expand_dims(data, axis=0)
-            self.slices_to_write.append(data)
+            self._slices_to_write.append(data)
             current_slice += 1
 
-            if current_slice % self.buffer_size == 0:
+            if current_slice % self._buffer_size == 0:
                 self._flush_buffer()
 
-    def __enter__(self) -> Generator[None, np.ndarray, None]:
-        gen = self._get_slice_generator()
-        # It is necessary to start the generator by sending "None"
-        gen.send(None)  # type: ignore
-        return gen
+    def send(self, value: np.ndarray) -> None:
+        self._generator.send(value)
+
+    def reset_offset(
+        self,
+        offset: Optional[Vec3IntLike] = None,  # deprecated, relative in current mag
+        relative_offset: Optional[Vec3IntLike] = None,  # in mag1
+        absolute_offset: Optional[Vec3IntLike] = None,  # in mag1
+        relative_bounding_box: Optional[NDBoundingBox] = None,  # in mag1
+        absolute_bounding_box: Optional[NDBoundingBox] = None,  # in mag1
+    ) -> None:
+        if getattr(self, "_slices_to_write", False):
+            self._flush_buffer()
```

Comment:
It's a bit weird to use getattr here because the propert should always exist. I assume it is because self._slices_to_write is initialized below the reset_offset call. However, it can probably simply be moved above that?

Reply:
Sure, that is much better.